src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -161,5 +161,23 @@ public sealed override void Visit(IValueVisitor visitor)
         {
             visitor.Visit(this);
         }
+
+        public virtual FormulaValue UpdateFields(IEnumerable<KeyValuePair<string, FormulaValue>> record)
```

Comment:
>FormulaValue [](http://example.com/codeflow?start=23&length=12)

This probably needs to be async. (ie, for dataverse, it will make a network call). 

Type should also be: 

Task<DValue<RecordValue>>